### PR TITLE
Adding THERMR toJSON capabilities.

### PR DESCRIPTION
**Pull request by jlconlin**

### CODE REVIEW

File: src/lipservice.hpp [hunk 1]
```diff
@@ -17,7 +17,7 @@
 namespace hana = boost::hana;
 
 #include "json.hpp"
-using json = nlohmann::json;
+// using json = nlohmann::json;
```
Comment (nathangibson14) [inline src/lipservice.hpp]: Commented code should be removed.

"This is the crappiest thing I've ever seen."  --Wim

File: src/lipservice/src/THERMR.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  JSON = {
+    { "tol", card4.tol.value },
+    { "emax", card4.emax.value }
+  };
+}
+
+inline void to_json( nlohmann::json& JSON, const THERMR& thermr ) {
+  JSON = thermr.card1;
+  JSON.update( thermr.card2 );
+  JSON[ "tempr" ] = thermr.card3;
```
Comment (nathangibson14) [inline src/lipservice/src/THERMR.hpp]: This syntax is different from the others, but doesn't need to be.  Could make a one-key dictionary above.

File: src/lipservice/THERMR/Card3/test/Card3.test.cpp [hunk 3]
```diff
@@ -13,19 +13,26 @@ SCENARIO( "Verifying THERMR Card3 input", "[THERMR], [Card3]" ){
...
         std::vector< double > refTemprs{ 293.6, 600, 1200 };
         THERMR::Card3 card3( iss, ntemp );
-        REQUIRE( refTemprs == card3.tempr.value );
+        CHECK( refTemprs == card3.tempr.value );
+
+        AND_THEN( "Card3 can be turned to JSON" ){
+          nlohmann::json refJSON = R"(
+          [ 293.6, 600, 1200 ]
+          )"_json;
+          CHECK( refJSON == nlohmann::json( card3 ) );
```
Comment (nathangibson14) [inline src/lipservice/THERMR/Card3/test/Card3.test.cpp]: If interface change is made, this test will need to be updated.

File: src/lipservice/THERMR/test/THERMR.test.cpp [hunk 4]
```diff
@@ -16,26 +16,52 @@ SCENARIO( "THERMR input",
...
-        REQUIRE( thermr.card3.tempr.value[1] == 450.0 );
+        CHECK( thermr.card3.tempr.value[0] == 350.0 );
+        CHECK( thermr.card3.tempr.value[1] == 450.0 );
 
-        REQUIRE( thermr.card4.tol.value  == 0.05 );
-        REQUIRE( thermr.card4.emax.value == 1.2 );
+        CHECK( thermr.card4.tol.value  == 0.05 );
+        CHECK( thermr.card4.emax.value == 1.2 );
+
+      AND_THEN( "RECONR can be turned to JSON" ){
```
Comment (nathangibson14) [inline src/lipservice/THERMR/test/THERMR.test.cpp]: Inconsistent indentation.

File: src/lipservice.hpp (hunk 1 above)
Comment (jlconlin) [reply src/lipservice.hpp]: Note: When quoting someone, you aught to use the @ sign to that everyone will know forever more @whaeck's feelings on the matter.

Comment (jlconlin) [reply src/lipservice.hpp]: Fixed

File: src/lipservice/src/THERMR.hpp (hunk 2 above)
Comment (jlconlin) [reply src/lipservice/src/THERMR.hpp]: Updated.

File: src/lipservice/THERMR/Card3/test/Card3.test.cpp (hunk 3 above)
Comment (jlconlin) [reply src/lipservice/THERMR/Card3/test/Card3.test.cpp]: done

File: src/lipservice/THERMR/test/THERMR.test.cpp (hunk 4 above)
Comment (jlconlin) [reply src/lipservice/THERMR/test/THERMR.test.cpp]: indented properly

**Comment by jlconlin** [comment]
Everything has been fixed. I'm building the tests now and will report whether or not they pass.

**Comment by jlconlin** [comment]
All the tests pass. Please approve if you do.

File: src/lipservice/THERMR/test/THERMR.test.cpp [hunk 5]
```diff
@@ -47,7 +73,52 @@ SCENARIO( "THERMR input",
...
+        CHECK( thermr.card2.mtref.value  == 221 );
+        CHECK( thermr.card2.iprint.value == 0 );
+	
+        CHECK( thermr.card3.tempr.value[0] == 350.0 );
+        CHECK( thermr.card3.tempr.value[1] == 450.0 );
+
+        CHECK( thermr.card4.tol.value  == 0.05 );
+        CHECK( thermr.card4.emax.value == 1.2 );
+
+      AND_THEN( "RECONR can be turned to JSON" ){
```
Comment (nathangibson14) [inline src/lipservice/THERMR/test/THERMR.test.cpp]: Improper indentation.

Comment (jlconlin) [reply src/lipservice/THERMR/test/THERMR.test.cpp]: Glad you found this. It is fixed and this Pull Request should be ready

**Comment by jlconlin** [comment]
Thanks @whaeck and @nathangibson14. Now let's look at #14

**Pull request merged**